### PR TITLE
Add support for specifying a log file from command line.

### PR DESCRIPTION
**Pull request by rictic**
Also improve documentation in the README.

 - [x] CHANGELOG.md has been updated

---
This change is [Reviewable](https://reviewable.io/reviews/polymer/polymer-editor-service/110)

### CODE REVIEW

File: src/language-server/logger.ts [hunk 1]
```diff
@@ -19,9 +19,20 @@ import Settings from './settings';
 
 export class Logger {
   private logFile: WriteStream|undefined = undefined;
+  private readonly logFileFromFlag: WriteStream|undefined;
```
Comment (TimvdLippe) [inline src/language-server/logger.ts]: This name feels weird, as it is not a flag but actually a stream. Why not `logFileStream`?

File: src/language-server/logger.ts [hunk 2]
```diff
@@ -19,9 +19,20 @@ import Settings from './settings';
...
   private settings: Settings|undefined;
   private queue: string[] = [];
-  constructor(private readonly connection: IConnection) {
+  constructor(
+      private readonly connection: IConnection,
+      logToFileFlag: string|undefined) {
```
Comment (TimvdLippe) [inline src/language-server/logger.ts]: I would make this an object with default value `{}`. That makes it more readable on the call-side, as in the test you had to add `undefined` as second argument which is ambigous. `new Logger(connection)` and `new Logger(connection, {logToFileFlag: true})` is better imo.

File: src/language-server/logger.ts (hunk 1 above)
Comment (rictic) [reply src/language-server/logger.ts]: It's not the flag itself, it's the file that the user asked us to log to based on a command line flag.

File: src/language-server/logger.ts (hunk 2 above)
Comment (rictic) [reply src/language-server/logger.ts]: For internal types I care much more about correctness than the convenience of optional properties and args, but the legibility gains of named args are nice here.

**Pull request merged**